ext/span.c

```diff
@@ -671,10 +672,7 @@ void ddtrace_drop_span(ddtrace_span_data *span) {
 }
 
 void ddtrace_serialize_closed_spans(zval *serialized) {
-    array_init(serialized);
-
-    // We need to loop here, as closing the last span root stack could add other spans here
-    while (DDTRACE_G(top_closed_stack)) {
+    if (DDTRACE_G(top_closed_stack)) {
```

Comment:
**question:** Is this comment not true anymore and is this related to not calling a garbage collection cycle?

Reply:
Yes, that's related to the garbage collection. The code in this function does not have any side effects by itself.